### PR TITLE
Improvement: send flow

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

Improvements based on @cjeria doc

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/Views/Settings/Contacts/ContactForm/index.js [hunk 1]
```diff
@@ -233,30 +283,78 @@ class ContactForm extends PureComponent {
...
+									multiline
+									editable={editable}
+									autoCapitalize={'none'}
+									autoCorrect={false}
+									onChangeText={this.onChangeMemo}
+									placeholder={strings('address_book.memo')}
+									placeholderTextColor={colors.grey100}
+									spellCheck={false}
+									numberOfLines={1}
+									onBlur={this.onBlur}
```
Comment (danjm) [inline app/components/Views/Settings/Contacts/ContactForm/index.js]: Is `this.onBlur` defined?

File: app/components/UI/Navbar/index.js [hunk 2]
```diff
@@ -199,6 +199,52 @@ export function getNavigationOptionsTitle(title, navigation) {
...
+	return {
+		title,
+		headerTitleStyle: {
+			fontSize: 20,
+			color: colors.fontPrimary,
+			...fontStyles.normal
+		},
+		headerTintColor: colors.blue,
+		headerLeft: (
+			<TouchableOpacity onPress={navigationPop} style={styles.backButton}>
```
Comment (danjm) [inline app/components/UI/Navbar/index.js]: Minor: We could pass `navigation.pop` directly to `onPress` to avoid creating the extra function.

**Review by danjm — CHANGES_REQUESTED** [review]
Left a couple comments. The only important one is the question about `this.onBlur` in `ContactForm/index.js`

File: app/components/Views/Settings/Contacts/ContactForm/index.js (hunk 1 above)
Comment (estebanmino) [reply app/components/Views/Settings/Contacts/ContactForm/index.js]: not defined, I can remove it

File: app/components/UI/Navbar/index.js (hunk 2 above)
Comment (estebanmino) [reply app/components/UI/Navbar/index.js]: it doesn't work that way

Comment (rickycodes) [reply app/components/UI/Navbar/index.js]: It could just be `onPress={e => navigation.pop()}`. I think having a separate function is fine though.

Comment (rickycodes) [reply app/components/UI/Navbar/index.js]: I see why it's a separate function now... doing what I proposed above results in a warning: `JSX props should not use arrow functions eslint(react/jsx-no-bind)`

**Comment by ibrahimtaveras00** [comment]
Some QA issues I saw from the mobile address book design QA doc:

**From 1.2 in doc**

When switching the account in the "From" field on the "send to" view, it doesn't detect the update 
For example, in this screencast Account 1 does have a balance greater than 0 ETH while Account 4 has 0 ETH, I correctly see the message displayed when FROM = Account 4, however, when I switch to Account 1, the message is still displayed

Seen here = https://recordit.co/rB9WamXrfk

image

Steps to reproduce:
This can be reproduced in 2 ways:
1) Start on an account that has 0 ETH then switch to an account that has some ETH
2) Start on an account that has ETH, then switch to an account that has 0 ETH

**From 3 in doc**

This issue is still happening "The last account used to send does not show up in my “Recents” list"

**From 6.1 in doc**

I do see the green check mark aligned next to the address when no name is present, however, when there is a name present, or has an ENS name, the green check mark is now in the middle, is this intended/how it should be? 

image


**From 7 in doc**

I do not see any updates from 7.1 to 7.5 
Are we ignoring this for now, or was this over looked? 

**From 8.1 in doc**

When you already have at least 1 contact, then add another contact, and then attempt to delete that newly added contact, the list does not update unless you leave the Contacts view by going back to settings and then coming back to the Contacts view

seen here = https://recordit.co/tZlryP2SeJ

**Other Issue**

I'm not sure which to associate this issue with in the doc, but nonetheless it is noticeable.

When I have an ENS name in my Recents list that's not saved yet in my contacts, and I tap on it to populate the To field, I do not see the option to add to my address book.

On the other hand, if I manually type the same exact ENS name, I do see the option.

Seen here = http://recordit.co/sEI6SxDyCJ

File: app/components/UI/Navbar/index.js (hunk 2 above)
Comment (estebanmino) [reply app/components/UI/Navbar/index.js]: is a separate function because it literally doesn't work, if I changed that the user couldn't go back

**Comment by estebanmino** [comment]
@ibrahimtaveras00  Issue 7 is https://github.com/MetaMask/metamask-mobile/issues/1334

File: app/components/Views/Settings/Contacts/ContactForm/index.js [hunk 3]
```diff
@@ -131,6 +133,11 @@ class ContactForm extends PureComponent {
 	componentDidMount = () => {
 		const { mode } = this.state;
 		const { navigation } = this.props;
+		// Workaround https://github.com/facebook/react-native/issues/9958
```
Comment (danjm) [inline app/components/Views/Settings/Contacts/ContactForm/index.js]: Interesting workaround. I guess we should check back on that react-native issue and remove the workaround once the underlying problem is resolved

**Review by danjm — APPROVED** [review]
All the code changes look good to me

**Comment by ibrahimtaveras00** [comment]
All fixes look good 👍 

Last issue I found is more of a question; do we want to have an ellipsis here @estebanmino @cjeria?

Screen Shot 2020-02-25 at 9 24 32 PM

Same as on the send view as well

Screen Shot 2020-02-25 at 9 27 31 PM

**Comment by cjeria** [comment]
> Last issue I found is more of a question; do we want to have an ellipsis here @estebanmino @cjeria?
Yes to ellipsis. Let's try to make it wide as possible.

**Comment by omnat** [comment]
QA: Recent addresses only the ones from address book, not the ones not saved. This logic needs changing. @danjm

**Comment by ibrahimtaveras00** [comment]
Ellipsis fix looks good 👍 

Screen Shot 2020-02-26 at 3 00 36 PM

**Review by ibrahimtaveras00 — APPROVED** [review]
Changes look good on both OS's QA Passed 👍

**Comment by cjeria** [comment]
Looking good! One additional change I'd like to suggest is to remove the text "No contacts in your address book" or at least use a grey color since I thought it looked like a text link, although I'm more in favor of just removing it altogether. 

image

**Pull request merged**